helper/testhelpers/testhelpers.go

```diff
@@ -262,215 +271,273 @@ func PassthroughWithLocalPathsFactory(ctx context.Context, c *logical.BackendCon
+	localopts.Logger = opts.Logger.Named("perf-pri-dr")
+	localopts.FirstCoreNumber += len(ret.PerfSecondaryCluster.Cores)
+	ret.PerfPrimaryDRCluster = MakeClusterAndStorage(t, InmemBackendSetup, conf, &localopts)
 
-	builder := &ReplicatedTestClustersBuilder{clusters: ret}
-	builder.setupFourClusterReplication(t)
+	localopts.Logger = opts.Logger.Named("perf-sec-dr")
+	localopts.FirstCoreNumber += len(ret.PerfPrimaryDRCluster.Cores)
+	ret.PerfSecondaryDRCluster = MakeClusterAndStorage(t, InmemBackendSetup, conf, &localopts)
 
-	// Wait until poison pills have been read
-	time.Sleep(45 * time.Second)
-	EnsureCoresUnsealed(t, ret.PerfPrimaryCluster)
-	EnsureCoresUnsealed(t, ret.PerfSecondaryCluster)
-	EnsureCoresUnsealed(t, ret.PerfPrimaryDRCluster)
-	EnsureCoresUnsealed(t, ret.PerfSecondaryDRCluster)
+	SetupFourClusterReplication(t, ret.PerfPrimaryCluster, ret.PerfSecondaryCluster, ret.PerfPrimaryDRCluster, ret.PerfSecondaryDRCluster)
 
 	return ret
 }
 
-type ReplicatedTestClustersBuilder struct {
-	clusters               *ReplicatedTestClusters
-	perfToken              string
-	drToken                string
-	perfSecondaryRootToken string
-	perfSecondaryDRToken   string
+func (r *ReplicatedTestClusters) SetupTwoClusterPerfReplication(t testing.T, maskSecondaryToken bool) {
+	SetupTwoClusterPerfReplication(t, r.PerfPrimaryCluster, r.PerfSecondaryCluster, maskSecondaryToken)
 }
 
-func SetupTwoClusterPerfReplication(t testing.T, pri, sec *vault.TestCluster) {
-	clusters := &ReplicatedTestClusters{
-		PerfPrimaryCluster:   pri,
-		PerfSecondaryCluster: sec,
+func SetupTwoClusterPerfReplication(t testing.T, pri, sec *vault.TestCluster, maskSecondaryToken bool) {
+	EnablePerfPrimary(t, pri)
+
+	var publicKey string
+	if maskSecondaryToken {
+		publicKey = generatePublicKey(t, sec)
 	}
-	builder := &ReplicatedTestClustersBuilder{clusters: clusters}
-	builder.setupTwoClusterReplication(t)
+	perfToken := GetPerformanceToken(t, pri, sec.ID, publicKey)
+
+	EnablePerformanceSecondary(t, perfToken, pri, sec, false, false)
 }
 
-func (r *ReplicatedTestClustersBuilder) setupTwoClusterReplication(t testing.T) {
-	t.Log("enabling perf primary")
-	r.enablePerfPrimary(t)
-	WaitForActiveNode(t, r.clusters.PerfPrimaryCluster)
-	r.getPerformanceToken(t)
-	t.Log("enabling perf secondary")
-	r.enablePerformanceSecondary(t)
+func GetDRReplicatedClusters(t testing.T, setup SetupGetter, conf *vault.CoreConfig, opts *vault.TestClusterOptions) *ReplicatedTestClusters {
+	clusters := PrepDRReplicatedClusters(t, setup, conf, opts)
+	SetupTwoClusterDRReplication(t, clusters.PerfPrimaryCluster, clusters.PerfPrimaryDRCluster, false)
+	return clusters
+}
+
+func (r *ReplicatedTestClusters) AddDRSecondaryCluster(t testing.T, setup SetupGetter, conf *vault.CoreConfig, opts *vault.TestClusterOptions) {
+	if setup == nil {
+		setup = InmemBackendSetup
+	}
+	opts = getClusterDefaultsOpts(t, opts, "perf-dr-pri")
+	opts.FirstCoreNumber += len(r.PerfPrimaryCluster.Cores)
```

Comment:
What if there are more than just a primary cluster? should it be 3 * number of clusters? (knowing that 3 is configurable)

Reply:
The only supported use case for AddPerfSecondaryCluster right now is a test or two that want to do stuff with a single cluster then add replication afterwards.  There are some other tests that create multiple perf secondarys, and I'd like to generalize this stuff to support them, but I think that'll be round two.  For now ReplicatedTestClusters is really only meant to be used for the 4-cluster, 2-perf, or 2-dr setups.  (Or the single-cluster setup prior to setting up replication.)